<!DOCTYPE html>
<html lang="en"><head>
<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/tabby.min.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.450">

  <meta name="author" content="Andrea Brizzi">
  <meta name="dcterms.date" content="2023-11-12">
  <title>HIV-phyloTSI: IAVI workshop</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="index_files/libs/revealjs/dist/theme/quarto.css">
  <link href="index_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="index_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <meta name="mermaid-theme" content="default">
  <script src="index_files/libs/quarto-diagram/mermaid.min.js"></script>
  <script src="index_files/libs/quarto-diagram/mermaid-init.js"></script>
  <link href="index_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">HIV-phyloTSI:<br>IAVI workshop</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Andrea Brizzi 
</div>
</div>
</div>

  <p class="date">2023-11-12</p>
</section>
<section>
<section id="intrahost-diversity-and-recency-of-infections" class="title-slide slide level1 center" data-background-color="#b5361c">
<h1>Intrahost diversity and recency of infections</h1>
<aside class="notes">
<p>40 minutes, this should be about background and things mentioned by Tanya Introduce power of</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<!--
## Recency of infection { background-color="#FFDFDD"}

* *What?*   
    Determining whether a person testing HIV positive has been infected recently or not.

. . .

* *Why ?* 
    + incidence estimation
    + study delays to diagnosis in subpopulations.
  
. . .

* *How ?*
  + longitudinal cohorts as long standard (expensive, time consuming)

. . .
-->
</section>
<section id="why-do-we-care-about-recency" class="slide level2" data-background-color="#FFDFDD">
<h2>Why do we care about recency?</h2>
<aside class="notes">
<ul>
<li>delays to diagnosis in sub-populations for 2nd one too</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment"><p>incidence estimation</p></li>
<li class="fragment"><p>identifying sub-populations sufferinng from recent infections are happening.</p></li>
<li class="fragment"><p>determining generation intervals.</p></li>
</ul>
<div class="fragment">
<p>but getting data is hard:</p>
<ul>
<li class="fragment"><p>longitudinal cohorts as gold standard</p></li>
<li class="fragment"><p>alternatives which are discussed in this section</p></li>
</ul>
</div>
</section>
<section id="what-is-recency-natural-history-of-hiv" class="slide level2 smaller" data-background-color="#FFDFDD" data-incremental="false">
<h2>What is recency? Natural history of HIV</h2>
<aside class="notes">
<ul>
<li>mention different stage classifications:Fiebing? …</li>
<li>describe natural history of HIV: complex interplay between VL and CD4 counts</li>
<li>mention “imbalance”: a very short dynamic period preceeds a long and static one. A very high VL may be indicative of a recent infection. However, a “medium” value does not help us discriminate/distinguish among other dates.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>

<img data-src="figures/HIV_natural_history.png" style="width:70.0%" class="r-stretch quarto-figure-center"><div class="fragment">
<p>dynamics define infection stages: <span class="citation" data-cites="centers_for_disease_control_and_prevention_cdc_revised_2014">(CDC: <a href="#/references" role="doc-biblioref" onclick="">2014</a>)</span> ; <span class="citation" data-cites="fiebig_dynamics_2003">(Fiebig et al. <a href="#/references" role="doc-biblioref" onclick="">2003</a>)</span></p>
<div class="footer">
<p>source: <a href="https://www.wikidoc.org/index.php/HIV_AIDS_natural_history,_complications,_and_prognosis">wikidoc</a></p>
</div>
</div>
</section>
<section id="lab-assays-to-identify-infection-stages" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>Lab assays to identify infection stages</h2>
<aside class="notes">
<ul>
<li>“lab” assay are based on the differences in internal “stages” of infection and immune response. For example a person may be suspected to be recent if first-level response antigens are present, but not other antigens more specific to HIV-1</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="columns">
<div class="column" style="width:50%;">
<aside class="notes">
<p>test by seeing whether blood react when exposed to specific antigens. 1. is based antigen test based on increase in antibodies over time 2. quality of antibody respons 3. proportion of HIV-specific IgG to total IgG.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<p>Based on the quality and type of immune response.</p>
<ul>
<li class="fragment"><strong>Less sensitive Enzyme Immunoassays</strong>, antigen test <span class="citation" data-cites="Janssen">(<a href="#/references" role="doc-biblioref" onclick=""><strong>Janssen?</strong></a>…)</span></li>
<li class="fragment"><strong>Lag-Avidity</strong> <span class="citation" data-cites="Duong2012">(<a href="#/references" role="doc-biblioref" onclick=""><strong>Duong2012?</strong></a>)</span></li>
<li class="fragment"><strong>Bed-CEIA</strong> <span class="citation" data-cites="Parek2011">(<a href="#/references" role="doc-biblioref" onclick=""><strong>Parek2011?</strong></a>)</span></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figures/HIV_serology.jpg" style="width:100.0%;height:100.0%"></p>
</div><div class="footer">
<p>source: <a href="https://depts.washington.edu/uwviro/hiv-diagnostic-and-molecular-testing/">UW medicine</a></p>
</div>
</div>
</section>
<section id="section" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2></h2>
<p>This is great, but specificity and sensitivity are hard to control:</p>
<ul>
<li class="fragment">individual level variation, confounding subtypes, ART status…</li>
</ul>

<img data-src="figures/HIV_subtypes.jpg" class="r-stretch quarto-figure-center"><div class="footer">
<p><span class="citation" data-cites="layendecker2013">(<a href="#/references" role="doc-biblioref" onclick=""><strong>layendecker2013?</strong></a>)</span></p>
</div>
<aside class="notes">
<p>Being affected by ART makes it harder to trust results in a world where ART is more available, subtype makes it harder to generalise by location</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="an-alternative-unit-of-time-genetic-diversity" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>An alternative unit of time: genetic diversity</h2>
<aside class="notes">
<ul>
<li>these assay generally target only one of the many facets of HIV response. I focused on antibody response and VL, but evolution dynamics bring up another source of information.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="columns">
<div class="column" style="width:40%;">
<p><img data-src="figures/moyo_identifying_figure1.png" style="width:100.0%;height:100.0%"></p>
<div class="footer">
<p><span class="citation" data-cites="moyo_identifying_2015">Moyo et al. (<a href="#/references" role="doc-biblioref" onclick="">2015</a>)</span> reviews molecular &amp; serological assays.</p>
</div>
</div><div class="column" style="width:60%;">
<aside class="notes">
<p>give detail on Ragonnet, still need a training set with recent and non-recent. + use of demographics.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment"><p>HIV infection is a multi-faceted phenomenon: immune response is just one facet.</p></li>
<li class="fragment"><p>HIV-1 diversity grows as a function on time since infection.</p></li>
<li class="fragment"><p>NGS is more informative than Sanger (<span class="citation" data-cites="carlisle_viral_2019">Carlisle et al. (<a href="#/references" role="doc-biblioref" onclick="">2019</a>)</span>)</p></li>
</ul>
</div>
</div>
</section>
<section id="case-study-recency-in-botswana" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>Case study: recency in Botswana</h2>
<aside class="notes">
<p>before moving to HIV-phyloTSI, I want to go through another application of NGS data in the context of estimating time since infection in an African setting. In particular, highlighting the strength and limitations of such studies will help us: 1. put in context the HIV-phyloTSI algorith and 2. understand its advantages and drawbacks</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<p><span class="citation" data-cites="ragonnet-cronin_human_2022">Ragonnet-Cronin et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span> developed recency classifier</p>
<ul>
<li class="fragment"><strong>end goal</strong>: characterizing undiagnosed cases in Botswana (<span class="citation" data-cites="bhebhe_epidemiological_2022">Bhebhe et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span>)</li>
<li class="fragment"><strong>data</strong>: demographic data, some testing data and individual level NGS</li>
<li class="fragment"><strong>how</strong>: train ML model on known recent and known chronic infections, and use it to classify unknown.</li>
</ul>
<div class="fragment">
<p>comments on model:</p>
<ol type="1">
<li class="fragment">still required cohort study, and <em>not directly generalisable</em> to other contexts.</li>
<li class="fragment"><em>simple measures of diversity</em>: entropy and mean pairwise distance</li>
<li class="fragment">these alone are <em>not enough</em> to provide good classification</li>
</ol>
</div>
</section>
<section id="case-study-recency-in-botswana-1" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>Case study: recency in Botswana</h2>
<div class="r-stack">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/ragonnet2022_fig2b.png" style="width:80.0%"></p>
</figure>
</div>
</div>
<div class="footer">
<p><span class="citation" data-cites="ragonnet-cronin_human_2022">Ragonnet-Cronin et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span></p>
</div>
</section>
<section id="hiv-phylo-tsi" class="slide level2" data-background-color="#FFDFDD">
<h2>HIV-phylo TSI</h2>
<aside class="notes">
<p>Addresses drawbacks of previous methods.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<p><span class="citation" data-cites="golubchik_hiv-phylotsi_2022">Golubchik et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span> tackles weaknesses of previous approach:</p>
<ol type="1">
<li class="fragment"><strong>Pre-trained</strong>: no need for expensive training data.</li>
<li class="fragment">Captures more complex <strong>measures of divergence</strong>.</li>
<li class="fragment">Comparatively <strong>low false recency rate</strong> using exclusively NGS data.</li>
</ol>
</section>
<section id="hiv-phylo-tsi-training-data" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>HIV-phylo TSI: training data</h2>
<p>sequences from PANGEA and BEEHIVE with known infection range.</p>

<img data-src="figures/HIVphyloTSI_trainingdata.png" style="width:100.0%" class="r-stretch quarto-figure-center"></section>
<section id="hiv-phylo-tsi-intrahost-diversity" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>HIV-phylo TSI: intrahost diversity</h2>
<p>choice of predictors based on viral diversification following infection.</p>
<ol type="1">
<li class="fragment">“Accumulated” mutations on codons.</li>
<li class="fragment">“Depth” of host-specific subtree in phylogeny. (LRTT)</li>
</ol>
<div class="r-stack">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/HIVphyloTSI_mafs.png" class="fragment" style="width:100.0%"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/phyloscanner_trees_genome.png" class="fragment" style="width:100.0%"></p>
</figure>
</div>
</div>
</section>
<section id="but-why-these-measures-of-intrahost-diversity" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>But why these measures of intrahost diversity?</h2>
<aside class="notes">
<p>250bp rolling mean: - regions along the genome where genetic divergence increases significantly over time - LRTT as “distance from founder” virus - MAF to measure accumulation of mutations</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>

<img data-src="figures/HIVphyloTSI_diversity_along_genome.png" style="width:100.0%" class="r-stretch"></section>
<section id="hiv-phylo-tsi-model" class="slide level2" data-background-color="#FFDFDD">
<h2>HIV-phylo TSI model</h2>
<p>different signal in different part of the genome. How to combine predictors?</p>
<ul>
<li class="fragment">take average over gene</li>
<li class="fragment">let the model pick up the informative covariates.</li>
</ul>
<div class="fragment">
<ul>
<li class="fragment"><p>Best performing set of features chosen through LOOCV</p></li>
<li class="fragment"><p>Regressor performs well as a classifier too, but FRR (~10%)</p></li>
</ul>
</div>
</section>
<section id="hiv-phylo-tsi-performances-bias" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>HIV-phylo TSI performances: bias</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ul>
<li class="fragment">bias is low in infections &lt; 9 years.</li>
<li class="fragment">range of individual-level errors can be quite wide.</li>
</ul>
<p>For applications:</p>
<ul>
<li class="fragment">estimates are not precise at the individuals level BUT</li>
<li class="fragment">there is signal in medians taken among subpopulation.</li>
</ul>
</div><div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/HIVphyloTSI_bias.png" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="hiv-phylotsi-performances-robustness-to-subtype" class="slide level2" data-background-color="#FFDFDD">
<h2>HIV-phyloTSI performances: robustness to subtype</h2>
<div class="columns">
<div class="column" style="width:35%;">
<p>model bias was close to 0 for all subtypes included in the dataset.</p>
</div><div class="column" style="width:65%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/HIVphyloTSI_subtype.png" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="hiv-phylotsi-performances-robustness-to-art" class="slide level2 smaller" data-background-color="#FFDFDD">
<h2>HIV-phyloTSI performances: robustness to ART (?</h2>
<div class="columns">
<div class="column" style="width:35%;">
<p>Some individuals in the training data reported prior ART use.</p>
<ul>
<li class="fragment"><p>followed pattern</p></li>
<li class="fragment"><p>BUT: small sample size</p></li>
<li class="fragment"><p>AND: no predictions for suppressed individuals.</p></li>
</ul>
</div><div class="column" style="width:65%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="figures/HIVphyloTSI_reportedart.png" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section></section>
<section>
<section id="hiv-phylotsi-bioinformatics-pipeline" class="title-slide slide level1 center" data-background-color="#1c9d7c">
<h1>HIV-PhyloTSI bioinformatics pipeline</h1>
<aside class="notes">
<p>30 minutes, talk about the pipeline. Break it in four parts.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="demistifying-the-process" class="slide level2" data-background-color="#A1FFDE">
<h2>Demistifying the process</h2>
<aside class="notes">
<p>objective is to describe and demistify the bioinformatics pipeline. - process of obtaining tsi estimates from blood samples of HIV positive individuals. - the pipeline is an essential part of the process, but we 1. no time and 2. implementation is technical and out-of-scope.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="cell" data-reveal="true">
<div class="cell-output-display">
<div>
<div>
<pre class="mermaid mermaid-js">flowchart LR 
START[(HIV+ blood samples)]
M1[???]
M2[???]
M3[???]
END[(TSI estimates)]

START --&gt; M1 --&gt; M2 --&gt; M3 --&gt; END
</pre>
</div>
</div>
</div>
</div>
<ul>
<li class="fragment">In this section, want to <strong>describe</strong> processes and <strong>data manipulation</strong> carried out by the pipeline.</li>
<li class="fragment"><strong>Not</strong> the actual <strong>implementation</strong> of the pipeline.</li>
</ul>
<p>Main steps:</p>
<ul>
<li class="fragment">Next Generation Sequencing (NGS)</li>
<li class="fragment">Constructing phylogenies <span class="fragment"><strong>MANY!</strong></span></li>
<li class="fragment">Analysing phylogenies</li>
</ul>
</section>
<section id="next-generation-sequencing" class="slide level2" data-background-color="#A1FFDE">
<h2>Next Generation Sequencing</h2>
<aside class="notes">
<p>NGS is the fundamental technology to capture and measure within-host diversity. It allows to sequence multiple viral strains from the same infected individuals, aligning them to the same reference sequence. As compared to the traditional Sanger sequencer, it allows to identify mutations, provide estimates of viral load and opens new doors. -&gt; fast, high-throughput, cost efficient</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment">simultaneous sequencing of multiple viral particles to capture within-host diversity</li>
<li class="fragment">PANGEA protocols: Gall <span class="citation" data-cites="gall_universal_2012">(<a href="#/references" role="doc-biblioref" onclick="">Gall et al. 2012</a>)</span> and Bonsall et al.<span class="citation" data-cites="bonsall_comprehensive_2018">(<a href="#/references" role="doc-biblioref" onclick="">Bonsall et al. 2018</a>)</span></li>
<li class="fragment">TSI pipeline starts with the outputs produced by <code>shiver</code> <span class="citation" data-cites="wymant_easy_2018">(<a href="#/references" role="doc-biblioref" onclick="">Wymant et al. 2018</a>)</span>: <code>*.bam</code>, <code>*.ref.fasta</code></li>
</ul>

<img data-src="figures/shiver.png" class="r-stretch quarto-figure-center"></section>
<section id="grouping-multiple-sequence-alignment" class="slide level2" data-background-color="#A1FFDE">
<h2>Grouping &amp; multiple sequence alignment</h2>
<aside class="notes">
<p>If there are multiple sequences per host, and there are many hosts for which we want predictions, the total number of sequences to consider increases very fast. Further, inferring tree structures becomes harder with the length of the sequences, and therefore we typically break the genome in chunks which are ~250 bp long. To avoid working with too many parts, we group individuals (or hosts) in groups of size 50, and aim to build one phylogeny per group, per 250 bp long genetic window. Building trees requires that we are able to map chunks of one virus’ genome to another virus’ genome (ie. comparing apple to apples.) Additionally, a set of “reference sequences” is used to make sure that alignments are “similar” across different groups of people. plot is too small right now</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment">Many sequences, many hosts, many genome windows <span class="fragment">-&gt; <strong>group by host!</strong></span></li>
<li class="fragment">aligning to compare “apples to apples”: MAFFT <span class="citation" data-cites="katoh_mafft_2002">(<a href="#/references" role="doc-biblioref" onclick="">Katoh 2002</a>)</span></li>
</ul>
<p><img data-src="figures/deepseqalign_workshop.png" class="fragment" data-fig-align="center"> ::: {.footer} The figure was generated with AliView :::</p>
</section>
<section id="inferring-phylogenies" class="slide level2" data-background-color="#A1FFDE">
<h2>Inferring phylogenies</h2>
<aside class="notes">
<p>Now that we obtained the alignments for every group of hosts and genome window, we can produce phylogenetic trees describing the evolutionary dependencies within- and between- hosts. This is done through IQTREE</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment">Phylogenies are made through IQTREE by group and window through IQTREE <span class="citation" data-cites="nguyen_iq-tree_2015">(<a href="#/references" role="doc-biblioref" onclick="">Nguyen et al. 2015</a>)</span></li>
</ul>

<img data-src="figures/ptyr1_InWindow_9075_to_9324_v2.treefile.png" class="r-stretch quarto-figure-center"><div class="footer">
<p>The image was generated with FigTree</p>
</div>
</section>
<section id="analysing-phylogenies" class="slide level2" data-background-color="#A1FFDE">
<h2>Analysing phylogenies</h2>
<aside class="notes">
<p>Once the trees are made, we need to programmatically study them and summarise them. We have seen in the previous section how features such as Longest Root to Tip distance are associated with time since infection and we would like to measure this along the entire genome. This is where <code>phyloscanner</code> comes into play. This is a tool which was principally developed to answers questions such as ‘whom infected whom’, but which produces a lot of useful summary information. These are included in the patient statistics file: <code>patStats</code>, which will be the input the HIV-phyloTSI algorithm later on. FIGURE: 5 small treefiles with a } bracket pointing to csv file</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment"><code>phyloscanner</code> (TOCITE) summarizes each tree through summary statistics:</li>
<li class="fragment"><code>patStats.csv</code>: contains LRTT, number of tips, etc…</li>
</ul>
</section>
<section id="minor-allele-frequencies" class="slide level2" data-background-color="#A1FFDE">
<h2>Minor Allele Frequencies</h2>
<aside class="notes">
<p>We almost have all the ingredients to cook Tanyas’ recipe: we are only missing Minor Allele Frequencies, which are a measure on the number of mutations that have appeared over time. Specifically, these are computed as the fraction of non-majority bases over all reads observeed at that position. Note that these measures are completely independent of the grouping/alignment procedures discussed previously, and can be independently obtained for each individuals sequence</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment"><span class="math inline">\(\text{MAF} = ( 1 - \text{proportion of majority bases}) / \text{depth}\)</span></li>
<li class="fragment">Evaluated at first 2 codos positions (<code>MAF12c</code>) and/or third codon position (<code>MAF3c</code>)</li>
</ul>

<img data-src="figures/maf.png" class="r-stretch quarto-figure-center"></section>
<section id="hiv-phylotsi" class="slide level2" data-background-color="#A1FFDE">
<h2>HIV-phyloTSI</h2>
<aside class="notes">
<p>Now that we have summarised the phylogenetic trees and evaluated diversity of the different reads, we have all the ingredients that are necessary to run HIV-phyloTSI. This is the section we specifically focus on during the practical.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment">described in <span class="citation" data-cites="golubchik_hiv-phylotsi_2022">Golubchik et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span>, <a href="https://github.com/BDI-pathogens/HIV-phyloTSI/tree/main">code</a></li>
<li class="fragment">Focus of the practical session,</li>
<li class="fragment">takes as input features of phylogenies and MAFs.</li>
<li class="fragment">returns a <code>csv</code> file with TSI estimates and some uncertainty range.</li>
</ul>
</section>
<section id="summary" class="slide level2 smaller" data-background-color="#A1FFDE">
<h2>Summary</h2>
<aside class="notes">
<p>So, to summarise, the differet steps are run according to this scheme.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="absolute" style="top: 20px; ">
<div class="cell" data-reveal="true">
<div class="cell-output-display">
<div>
<div>
<pre class="mermaid mermaid-js">graph LR
    subgraph S1["Primary Data Patients"]
        direction TB
        I01[(*.bam)] 
        I02[(*ref_fasta)] 
        I02b[(*BaseFreqs_WithHXB2.csv)]
    end

    subgraph S2["Reference Sequences"]
        I03[(*.fasta)] 
    end

    subgraph S3["Multiple Sequence Alignment"]
        P1(mafft)
        O1[(*.fasta)]
    end

    subgraph S4["Constructing Phylogenies"]
        direction TB
        P2(IQTREE)
        O2[( *.iqtree )]
    end
    
    subgraph S2b["Computing MAFs"]
      direction TB
      P2b(script)
      O2b[(maf.csv)]
    end

    subgraph S5["Analysing phylogenies"]
        direction TB
        P3( PhyloscannerR ) 
        O3[(*PatStats.csv)]
    end

    subgraph S6["Obtain TSI estimates"]
        direction TB
        P4( HIV-phyloTSI ) 
        O4[(TSI.csv)]
    end
    
    classDef Red fill:#F84E68
    class S3,S4 Red
    classDef Orange fill:orange
    class S5,S6 Orange

I02 --&gt; P1
I01 --&gt; P1 
I02b --&gt; P2b --&gt; O2b --&gt; P4
I03 --&gt; P1
    P1 --&gt; O1
O1 --&gt; P2 
    P2 --&gt; O2
O2 --&gt; P3 
    P3 --&gt; O3
O3 --&gt; P4 
    P4 --&gt; O4
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="absolute" style="bottom: 55px; ">
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>Steps run for each group and window are shown in red, while those running by group in orange</p>
</div>
</div>
</div>
</div>
</section>
<section id="but-what-is-hiv-phylotsi-doing" class="slide level2" data-background-color="#A1FFDE">
<h2>But what is HIV-phyloTSI doing?</h2>
<aside class="notes">
<p>Before we move to the practical, it may be worth to spend a couple minutes to explain what HIV-phyloTSI is doing under the hood: what type of ML model it is using and how it was trained.</p>
<p>random forest algorithm: little intuition Good thing is we do not need to understand the little details as long as we use the code provided.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li class="fragment"><strong>ML algorithm</strong>, in particular Random Forest.</li>
</ul>
</section></section>
<section>
<section id="hiv-phylotsi-obtaining-tsi-estimates" class="title-slide slide level1 center" data-background-color="#369cc9">
<h1>HIV-PhyloTSI obtaining TSI estimates</h1>
<aside class="notes">
<p>40 minutes, groups of 4</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plan" class="slide level2" data-background-color="#DEF1FF">
<h2>Plan</h2>
<ul>
<li class="fragment">divide in group of 4, each goup with a machine running Linux or MacOS</li>
<li class="fragment">introduction of sample dataset</li>
<li class="fragment">installing dependencies</li>
<li class="fragment">running hiv-phyloTSI</li>
<li class="fragment">if there is time, visualising outputs in R.</li>
</ul>
</section>
<section id="preliminary-tools" class="slide level2" data-background-color="#DEF1FF">
<h2>Preliminary Tools</h2>
<aside class="notes">
<p>-&gt; open Terminal macOS : Click the Launchpad icon in the Dock, type Terminal in the search field, then click Terminal. -&gt; Linux: Typical shortcut: Ctrl+Alt+T</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<p>There are two tools that will make our lives easier:</p>
<ul>
<li class="fragment"><strong>Terminal access</strong> allows to run programs and give instructions to computer. &lt;&gt; (<code>Ctrl+Alt+T</code> on Linux, <code>Terminal</code> on MacOS, see <a href="https://learn.microsoft.com/en-us/windows/wsl/install">WSL</a> for Windows ).</li>
<li class="fragment"><strong>Conda</strong> allows to download the exact requirements needed for HIV-phyloTSI. Installation instructions can be found <a href="https://conda.io/projects/conda/en/latest/user-guide/install/index.html">here</a></li>
</ul>
</section>
<section id="installing-dependencies" class="slide level2 smaller" data-background-color="#DEF1FF">
<h2>Installing dependencies</h2>
<p>The ‘ingredients’ to run the analyses are: the <strong>ML algorithm</strong>; the <strong>data</strong> and the <strong>code dependencies</strong>. The below code chunk allows you to download everything that is needed for the analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># change directory to where you want to install HIV-phyloTSI repo</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="bu">cd</span> <span class="va">$HOME</span>/git            <span class="co"># this is where I install git packages </span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># cd $HOME &amp;&amp; mkdir git &amp;&amp; cd git</span></span>
<span id="cb1-4"><a href="#cb1-4"></a></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co"># clone directories necessary to run the analysis</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co"># BDIs code and workshop materials</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">git</span> clone git@github.com/BDI-pathogens/HIV-phyloTSI.git</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">git</span> clone git@github.com/abriz97/HIV-phyloTSI-workshops.git</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co"># store paths to 2 above directories</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="va">DIR_WORKSHOP</span><span class="op">=</span><span class="st">"</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span><span class="st">/HIV-phyloTSI-workshops"</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="va">DIR_PROGRAM</span><span class="op">=</span> <span class="st">"</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span><span class="st">/HIV-phyloTSI"</span></span>
<span id="cb1-12"><a href="#cb1-12"></a></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="co"># install python dependencies for HIV-phyloTSI and load the environment</span></span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="ex">conda</span> env create <span class="at">-f</span> HIV-phyloTSI-workshops/hivphylotsi.yml</span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="ex">conda</span> activate hivphylotsi</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>When interacting with a terminal, learning the power of the <code>$</code> operator is key. The operator allows to evaluate variables (eg. <code>$HOME</code>) or to evaluate commands surrounded by brackets ( eg. <code>$(pwd)</code>).</p>
</div>
</div>
</div>
</section>
<section id="running-the-algorithm" class="slide level2 smaller" data-background-color="#DEF1FF">
<h2>Running the algorithm</h2>
<p>Once all the ingredients are there, we can start cooking. It is relatively simple to run the analyses, even though we need to be precise in the way we specify the paths to the input data.</p>
<div class="sourceCode" id="cb2" data-code-line-numbers="|2-3|4-6|8-9"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Run HIV-phyloTSI on input data.</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="ex">python</span> <span class="va">$DIR_PROGRAM</span>/HIV-phyloTSI.py <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>    <span class="at">-d</span> <span class="va">$DIR_PROGRAM</span>/Model <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>    <span class="at">-p</span> <span class="va">$DIR_WORKSHOP</span>/HIV-phyloTSI-workshops/input/ptyr1_patStats.csv <span class="dt">\ </span></span>
<span id="cb2-5"><a href="#cb2-5"></a>    <span class="ex">-m</span> <span class="va">$DIR_WORKSHOP</span>/HIV-phyloTSI-workshops/input/phsc_input_samples_maf.csv <span class="dt">\ </span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    <span class="ex">-o</span> <span class="va">$DIR_WORKSHOP</span>/HIV-phyloTSI-workshops/output/ptyr1_tsi_workshop.csv</span>
<span id="cb2-7"><a href="#cb2-7"></a>  </span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># print header of output to make sure it exists:</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="fu">head</span> <span class="va">$DIR_WORKSHOP</span>/HIV-phyloTSI-workshops/output/ptyr1_tsi_workshop.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>The first 2 lines point to <code>$DIR_PROGRAM</code> because they refer to the code we want to use. On the other hand, the bottom 3 lines refer to the input data and output paths, and this is why they point to <code>$DIR_WORKSHOP</code>.</p>
</div>
</div>
</div>
</section>
<section id="visualising-results" class="slide level2 smaller noincremental" data-background-color="#DEF1FF">
<h2>Visualising results</h2>
<p>As a first step, it is helpful to visualize the results and check whether they make sense. I provide some <strong>R functions and scripts</strong> to help the process, which <strong>can be found</strong> in the github <strong>repository</strong>:</p>
<ul>
<li class="fragment">script: <code>$DIR_WORKSHOP/src/workshop_analyses.R</code></li>
<li class="fragment">functions: <code>$DIR_WORKSHOP/src/R/workshop_R_helpers.R</code></li>
</ul>
<p>Again, we can use conda to install the necessary packages:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># install few R dependencies for visualisation and load the environment</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="bu">cd</span> <span class="va">$DIR_WORKSHOP</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="ex">conda</span> env create <span class="at">-f</span> workshopR.yml</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="ex">conda</span> activate workshopR</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I will be showing snippets of the above code together with the plots they produce. <br> You can reproduce the steps by opening up the script in RStudio.</p>
</section>
<section id="predictors-used" class="slide level2 smaller" data-background-color="#DEF1FF">
<h2>Predictors used</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>dtsi  <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="fu">file.path</span>(git_root, <span class="st">"data/ptyr1_tsi.csv"</span>))</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="fu">plot_predictors_from_tsi_output</span>(dtsi, <span class="at">exclude=</span><span class="st">"dual"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="figures/predictors.png" class="r-stretch quarto-figure-center"></section>
<section id="evaluation-on-seroconverters" class="slide level2 smaller" data-background-color="#DEF1FF">
<h2>Evaluation on seroconverters</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>dall  <span class="ot">&lt;-</span> <span class="fu">merge</span>(dtsi, ddates)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="fu">plot_cross_interval_tsisero</span>(dall, <span class="at">sqroot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="figures/crosscheck.png" class="r-stretch quarto-figure-center"></section>
<section id="histogram-of-estimates-tsis" class="slide level2 smaller" data-background-color="#DEF1FF">
<h2>Histogram of estimates TSIs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>p_hist <span class="ot">&lt;-</span> <span class="fu">plot_histogram_tsi</span>(dtsi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="figures/histogram_phyloTSI.png" class="r-stretch quarto-figure-center"></section></section>
<section>
<section id="possible-follow-up-analyses" class="title-slide slide level1 center" data-background-color="#3a507f">
<h1>Possible follow-up analyses</h1>
<aside class="notes">
<p>30 minutes, really stress meant at population level.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="plan-1" class="slide level2" data-background-color="#D9E2FF">
<h2>Plan</h2>
<ul>
<li class="fragment"><p>Simple stratification by age, sex, and other covariates of interest.</p></li>
<li class="fragment"><p>what NOT to do: changes in TSI by year of infection</p></li>
<li class="fragment"><p>How I personally usesd them for the MM paper.</p></li>
<li class="fragment"><p>mention Hinch’s work on Generation intervals.</p></li>
</ul>
</section>
<section id="simple-ideas" class="slide level2">
<h2>simple ideas:</h2>
<ul>
<li class="fragment">aggregate by (wide) age groups and compare medians (maybe bootstap to show uncertainty of medians)</li>
</ul>
</section>
<section id="simulation-study" class="slide level2">
<h2>Simulation study:</h2>
<aside class="notes">
<p>Blue: true recents Orange: estimated recents Green: false recents: ie estimated recents but chronic bar at 2%</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<p>In simulated settings, the prevalence of recent infections is well estimated, despite at least 2% of recent classifications are wrong. <img data-src="figures/HIVphyloTSI_simulatedcasestudy.png" style="width:100.0%"></p>
<div class="footer">
<p><span class="citation" data-cites="golubchik_hiv-phylotsi_2022">Golubchik et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span></p>
</div>
</section>
<section id="pop-art-ancillary" class="slide level2">
<h2>Pop ART ancillary</h2>
<div class="columns">
<div class="column" style="width:30%;">
<p><img data-src="figures/HIVphyloTSI_popart_seroconverters.png" style="height:90.0%"></p>
</div><div class="column" style="width:70%;">
<p><img data-src="figures/HIVphyloTSI_popart.png" style="height:80.0%"></p>
</div>
</div>
</section>
<section id="hptn-application" class="slide level2">
<h2>HPTN application:</h2>
</section></section>
<section id="references" class="title-slide slide level1 smaller scrollable">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-bhebhe_epidemiological_2022" class="csl-entry" role="listitem">
Bhebhe, Lynnette, Sikhulile Moyo, Simani Gaseitsiwe, Molly Pretorius-Holme, Etienne K. Yankinda, Kutlo Manyake, Coulson Kgathi, et al. 2022. <span>“Epidemiological and Viral Characteristics of Undiagnosed <span>HIV</span> Infections in <span>Botswana</span>.”</span> <em>BMC Infectious Diseases</em> 22 (1): 710. <a href="https://doi.org/10.1186/s12879-022-07698-4">https://doi.org/10.1186/s12879-022-07698-4</a>.
</div>
<div id="ref-bonsall_comprehensive_2018" class="csl-entry" role="listitem">
Bonsall, David, Tanya Golubchik, Mariateresa De Cesare, Mohammed Limbada, Barry Kosloff, George MacIntyre-Cockett, Matthew Hall, et al. 2018. <span>“A Comprehensive Genomics Solution for <span>HIV</span> Surveillance and Clinical Monitoring in a Global Health Setting.”</span> Preprint. Genomics. <a href="https://doi.org/10.1101/397083">https://doi.org/10.1101/397083</a>.
</div>
<div id="ref-carlisle_viral_2019" class="csl-entry" role="listitem">
Carlisle, Louisa A, Teja Turk, Katharina Kusejko, Karin J Metzner, Christine Leemann, Corinne D Schenkel, Nadine Bachmann, et al. 2019. <span>“Viral <span>Diversity</span> <span>Based</span> on <span>Next</span>-<span>Generation</span> <span>Sequencing</span> of <span>HIV</span>-1 <span>Provides</span> <span>Precise</span> <span>Estimates</span> of <span>Infection</span> <span>Recency</span> and <span>Time</span> <span>Since</span> <span>Infection</span>.”</span> <em>The Journal of Infectious Diseases</em> 220 (2): 254–65. <a href="https://doi.org/10.1093/infdis/jiz094">https://doi.org/10.1093/infdis/jiz094</a>.
</div>
<div id="ref-centers_for_disease_control_and_prevention_cdc_revised_2014" class="csl-entry" role="listitem">
Centers for Disease Control and Prevention (CDC). 2014. <span>“<a href="https://www.ncbi.nlm.nih.gov/pubmed/24717910">Revised Surveillance Case Definition for <span>HIV</span> Infection–<span>United</span> <span>States</span>, 2014</a>.”</span> <em>MMWR. Recommendations and Reports: Morbidity and Mortality Weekly Report. Recommendations and Reports</em> 63 (RR-03): 1–10.
</div>
<div id="ref-fiebig_dynamics_2003" class="csl-entry" role="listitem">
Fiebig, Eberhard W, David J Wright, Bhupat D Rawal, Patricia E Garrett, Richard T Schumacher, Lorraine Peddada, Charles Heldebrant, et al. 2003. <span>“Dynamics of <span>HIV</span> Viremia and Antibody Seroconversion in Plasma Donors: Implications for Diagnosis and Staging of Primary <span>HIV</span> Infection.”</span> <em>AIDS</em> 17 (13): 1871–79. <a href="https://doi.org/10.1097/00002030-200309050-00005">https://doi.org/10.1097/00002030-200309050-00005</a>.
</div>
<div id="ref-gall_universal_2012" class="csl-entry" role="listitem">
Gall, Astrid, Bridget Ferns, Clare Morris, Simon Watson, Matthew Cotten, Mark Robinson, Neil Berry, Deenan Pillay, and Paul Kellam. 2012. <span>“Universal <span>Amplification</span>, <span>Next</span>-<span>Generation</span> <span>Sequencing</span>, and <span>Assembly</span> of <span>HIV</span>-1 <span>Genomes</span>.”</span> <em>Journal of Clinical Microbiology</em> 50 (12): 3838–44. <a href="https://doi.org/10.1128/JCM.01516-12">https://doi.org/10.1128/JCM.01516-12</a>.
</div>
<div id="ref-golubchik_hiv-phylotsi_2022" class="csl-entry" role="listitem">
Golubchik, Tanya, Lucie Abeler-Dörner, Matthew Hall, Chris Wymant, David Bonsall, George Macintyre-Cockett, Laura Thomson, et al. 2022. <span>“<span>HIV</span>-<span class="nocase">phyloTSI</span>: <span>Subtype</span>-Independent Estimation of Time Since <span>HIV</span>-1 Infection for Cross-Sectional Measures of Population Incidence Using Deep Sequence Data.”</span> Preprint. HIV/AIDS. <a href="https://doi.org/10.1101/2022.05.15.22275117">https://doi.org/10.1101/2022.05.15.22275117</a>.
</div>
<div id="ref-katoh_mafft_2002" class="csl-entry" role="listitem">
Katoh, K. 2002. <span>“<span>MAFFT</span>: A Novel Method for Rapid Multiple Sequence Alignment Based on Fast <span>Fourier</span> Transform.”</span> <em>Nucleic Acids Research</em> 30 (14): 3059–66. <a href="https://doi.org/10.1093/nar/gkf436">https://doi.org/10.1093/nar/gkf436</a>.
</div>
<div id="ref-moyo_identifying_2015" class="csl-entry" role="listitem">
Moyo, Sikhulile, Eduan Wilkinson, Vladimir Novitsky, Alain Vandormael, Simani Gaseitsiwe, Max Essex, Susan Engelbrecht, and Tulio De Oliveira. 2015. <span>“Identifying <span>Recent</span> <span>HIV</span> <span>Infections</span>: <span>From</span> <span>Serological</span> <span>Assays</span> to <span>Genomics</span>.”</span> <em>Viruses</em> 7 (10): 5508–24. <a href="https://doi.org/10.3390/v7102887">https://doi.org/10.3390/v7102887</a>.
</div>
<div id="ref-nguyen_iq-tree_2015" class="csl-entry" role="listitem">
Nguyen, Lam-Tung, Heiko A. Schmidt, Arndt Von Haeseler, and Bui Quang Minh. 2015. <span>“<span>IQ</span>-<span>TREE</span>: <span>A</span> <span>Fast</span> and <span>Effective</span> <span>Stochastic</span> <span>Algorithm</span> for <span>Estimating</span> <span>Maximum</span>-<span>Likelihood</span> <span>Phylogenies</span>.”</span> <em>Molecular Biology and Evolution</em> 32 (1): 268–74. <a href="https://doi.org/10.1093/molbev/msu300">https://doi.org/10.1093/molbev/msu300</a>.
</div>
<div id="ref-ragonnet-cronin_human_2022" class="csl-entry" role="listitem">
Ragonnet-Cronin, Manon, Tanya Golubchik, Sikhulile Moyo, Christophe Fraser, Max Essex, Vlad Novitsky, and Erik Volz. 2022. <span>“Human <span>Immunodeficiency</span> <span>Virus</span> (<span>HIV</span>) <span>Genetic</span> <span>Diversity</span> <span>Informs</span> <span>Stage</span> of <span>HIV</span>-1 <span>Infection</span> <span>Among</span> <span>Patients</span> <span>Receiving</span> <span>Antiretroviral</span> <span>Therapy</span> in <span>Botswana</span>.”</span> <em>The Journal of Infectious Diseases</em> 225 (8): 1330–38. <a href="https://doi.org/10.1093/infdis/jiab293">https://doi.org/10.1093/infdis/jiab293</a>.
</div>
<div id="ref-wymant_easy_2018" class="csl-entry" role="listitem">
Wymant, Chris, François Blanquart, Tanya Golubchik, Astrid Gall, Margreet Bakker, Daniela Bezemer, Nicholas J Croucher, et al. 2018. <span>“Easy and Accurate Reconstruction of Whole <span>HIV</span> Genomes from Short-Read Sequence Data with Shiver.”</span> <em>Virus Evolution</em> 4 (1): vey007. <a href="https://doi.org/10.1093/ve/vey007">https://doi.org/10.1093/ve/vey007</a>.
</div>
</div>
<div class="footer footer-default">
<p><a href="https://www.github.com/abriz97/HIV-phyloTSI-workshops">Code and slides</a></p>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="index_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="index_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="index_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="index_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="index_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="index_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="index_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="index_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="index_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="index_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>